model/client4.go

```diff
@@ -3966,6 +3966,27 @@ func (c *Client4) GetPostsByIds(postIds []string) ([]*Post, *Response, error) {
 	return list, BuildResponse(r), nil
 }
 
+// GetEditHistoryForPost gets a list of posts by taking a post ids
+func (c *Client4) GetEditHistoryForPost(postId string) ([]*Post, *Response, error) {
+	js, err := json.Marshal(postId)
+	if err != nil {
+		return nil, nil, NewAppError("GetEditHistoryForPost", "api.marshal_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
+	r, err := c.DoAPIPost(c.postRoute(postId)+"/edit_history", string(js))
```

Comment:
BTW, the reason you get error in `api4/post_test.go:2986` is you need to make a `GET` request because the endpoint is defined that way. https://github.com/mattermost/mattermost-server/blob/85488813fe959f600b6c3cdfbfbdb928d3b8bfab/api4/post.go#L25

Reply:
🤦 I knew that I was missing something basic. Thanks for the support 😃